aiida/orm/entities.py

```diff
@@ -118,15 +133,16 @@ def get(self, **filters):
 
         :return: the entry
         """
-        res = self.find(filters=filters)
-        if not res:
+        res = self.query(filters=filters, limit=1)
```

Comment:
This is incorrect the `limit=1` will now hide if there are multiple objects, i.e. your exception in line 141 will never be hit

Reply:
By the way, check `QueryBuilder.first`, probably that already includes all the necessary logic to do what is done here to check for a uniquely resolvable identifier.

Reply:
> By the way, check `QueryBuilder.first`, probably that already includes all the necessary logic to do what is done here to check for a uniquely resolvable identifier.

Good idea.

Hmm, concerning my longer comment above. Shouldn't a filter then be made in the `Entity` class' `get` method that calls `Collection.get()`?

Reply:
Simply using the `.one()` method from `QueryBuilder`, which indeed has the logic already in place.

Concerning the `Entity.get()` filter, I guess it doesn't make sense, but maybe the method name should be changed to show that _only_ one entity is expected to be found and returned. E.g., `get_one()` or something?

Reply:
Of course I meant `one`, my bad. The name of `get` is actually modeled after [Django](https://docs.djangoproject.com/en/2.2/topics/db/queries/#retrieving-a-single-object-with-get) and is the common name for this type of query

Reply:
> Of course I meant `one`, my bad. The name of `get` is actually modeled after [Django](https://docs.djangoproject.com/en/2.2/topics/db/queries/#retrieving-a-single-object-with-get) and is the common name for this type of query

Doesn't make it clearer - only makes it traditional and conservative ... :unamused: :sweat_smile: Nah, I see your point. Let's keep it like this, but just use `one()` (as is now committed) and perhaps consider updating the method documentation for `Entity.get()` and perhaps others where/if needed?